File: RedSheet.DbEntities/Models/NanoTheirBatna.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("NanoTheirBatnas", Schema = "dbo")]
	[Serializable]
	public partial class NanoTheirBatna  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int NanoTheirBatnaId { get; set; } 
 
	[Required] 
	[MaxLength(400)]
	public string NanoTheirBatnaValue { get; set; } 
 
	[RelationshipTableAttribue("ProjectModules","dbo")]
	//Relationships
	public int ProjectModuleId { get; set; } 
 
	[ForeignKey("ProjectModuleId")]
	public ProjectModule ProjectModule { get; set; } 

	public NanoTheirBatna()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/nano-their-batna.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule,  vNanoTheirBatnaRecord  } from './'
export class NanoTheirBatna {
    constructor(nanoTheirBatna?: NanoTheirBatna  | vNanoTheirBatnaRecord )  {
        let properties = [ "nanoTheirBatnaId", "nanoTheirBatnaValue", "projectModuleId",];
        for (let property of properties)
            if (nanoTheirBatna && nanoTheirBatna[property])
                this[property] = nanoTheirBatna[property];
    }
 
	nanoTheirBatnaId : number =   0 ;
 
    @required()
    @maxLength(400)
	nanoTheirBatnaValue : string =   undefined;
 
    @range(0,2147483647)
	projectModuleId : number =   undefined;
	projectModule : ProjectModule  ;


}
